modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala

```diff
@@ -446,14 +443,12 @@ object TcsControllerEpics extends TcsController {
     demand.gds.pwfs2OrAowfs.toOption.filter(_.detector === GuiderSensorOn).as(GaosGuideOn)
   ).collect{ case Some(x) => x }
 
-  def pauseResumeGaos(gaos: Option[Either[Altair[IO], Gems[IO]]], current: EpicsTcsConfig, demand: TcsConfig)
-  : SeqAction[PauseResume[IO]] = (gaos, demand.gaos).mapN {
-    case (Left(g), c)  => SeqActionF.embed[IO, Gaos.PauseResume[IO]](
-      g.pauseResume(c, calcAoPauseConditions(current, demand), calcAoResumeConditions(current, demand)))
-    case (Right(_), _) => SeqAction.fail[PauseResume[IO]](
-      SeqexecFailure.Unexpected("GeMS not supported"))
-    case _             => SeqAction(PauseResume[IO](None, None))
-  }.getOrElse(SeqAction(PauseResume(None, None)))
+  def pauseResumeGaos[F[_]: Sync ](gaos: Option[Either[Altair[F], Gems[F]]], current: EpicsTcsConfig, demand: TcsConfig)
+  : F[PauseResume[F]] = (gaos, demand.gaos).mapN {
+    case (Left(g), c)  => g.pauseResume(c, calcAoPauseConditions(current, demand), calcAoResumeConditions(current, demand))
+    case (Right(_), _) => Sync[F].raiseError[PauseResume[F]](SeqexecFailure.Unexpected("GeMS not supported"))
+    case _             => Sync[F].pure((PauseResume[F](None, None)))
```

Comment:
You can also do `PauseResume[F](none, none).pure[F]`, you may even be able to drop some of the `F` if the compiler lets you